tools/topology/m4/dai.m4

```diff
@@ -44,7 +44,8 @@ define(`W_DAI_OUT',
 `SectionVendorTuples."'N_DAI_OUT($2)`_tuples_comp_str" {'
 `	tokens "sof_comp_tokens"'
 `	tuples."string" {'
-`		SOF_TKN_COMP_FORMAT'	STR($4)
+`		SOF_TKN_COMP_FORMAT'    `ifelse($1, "HDA",
+		`ifelse($4, `s24le', `"s32le"', STR($4))', `STR($4)')'
```

Comment:
@RanderWang why is this needed? Why are you overriding the s24le set in the top-level m4 to s32le?

Reply:
Doesnt this change make the other changes redundant?

Reply:
volume component in FW would check source and sink format to select conversion function, so
I need to change dai format. 

Yes, you are right if there is a volume at the end of pipeline. I am try to figure out how to know whether there is a volume at the end of pipeline and if I know it, I don't need to add volume again

Reply:
@RanderWang I still see Intel specific logic in generic code. Why don't you take each pipeline in machine.m4 and use a volume pipeline(instead of passthrough) for every pipeline that has this requirement.  Please see the diff I commented yesterday.